accepted/2020/workloads/workload-manifest.md

```diff
@@ -0,0 +1,430 @@
+
+To make this relationship more explicit, the SDK band for a manifest is encoded into the manifest’s ID e.g. `Microsoft.NET.Sdk.Android.SdkManifest-5.0.200`. The “version” of a manifest package can then be then simply a monotonic integer value.
+
+## Pack Versioning
+
+A manifest contains a list of pack IDs and their corresponding versions. A single version is defined for each pack: the most up-to-date version of that pack that is expected to be used with that SDK band.
+
+This intentionally separates pack versions from component definitions. A pack may be included in multiple workloads, and this separation ensures that all workloads get a consistent version of each pack. There’s also a semantic separation: the workload definition list describes which packs are included in each workload, while the pack version list describes the set of pack versions that are supported together.
+
+There are some cases where a single workload may want to contain side-by-side multiple versions of a single pack, or multiple components may want different versions of a single pack. This may apply, for example, to major/minor runtime versions, For these cases an aliasing mechanism is provided: a meaningful alias may be defined and mapped to a different version of a pack that is already included in the list. For example, `Foo-1.x` could be defined as an alias to version `1.0.5` of pack `Foo`, while `Foo` itself could be mapped to `2.1.4`. This is similar to how Linux libraries support side-by-side.
+
+# Pack Kinds
+
+The following kind of packs are permitted in manifests: targeting and runtime packs, SDK packs, library packs, template packs, and tools packs.
+
+All packs are expected to be installed into the `dotnet` folder.
+
+## Targeting and Runtime Packs
+
+The [targeting and runtime packs](https://github.com/dotnet/designs/pull/50) introduced in .NET Core 3.0 are permitted in workloads. Targeting packs contain the framework assemblies that a project compiles against at build time, and runtime packs contain the framework and runtime binaries used to execute the project. They install into `dotnet/packs/{pack-id}/{pack-version}` and are resolved at build time using `FrameworkReference` MSBuild items.
+
+Framework references are expected to be added to a project automatically by the manifest’s MSBuild targets based on the TFM. There will be a mechanism that allows the targets to omit the pack version from the framework reference. Targets may add multiple targeting or runtime pack references, for example when building a `net5.0-ios` project they could add a reference to a pack containing the `net5.0` reference assemblies and a pack containing the reference assemblies for the iOS platform bindings.
```

Comment:
Maybe a little in the weeds, but it might be helpful to explain that some fx references may be handled by additional mechanisms. Here, I'm thinking of `UseWinforms` and `UseWPF`.

Reply:
> There will be a mechanism that allows the targets to omit the pack version from the framework reference.

The pack version is already not normally specified on the `FrameworkReference` item.  It is specified in the `KnownFrameworkReference` which gets matched up with the `FrameworkReference` in the ProcessFrameworkReferences task.  The `KnownFrameworkReference` items are defined in Microsoft.NETCoreSdk.BundledVersions.props.

I was imagining that each workload would have it's own bundled versions file with the targeting and runtime pack version information.  This would be a duplicate of the data in the manifest.  It sounds like you're suggesting that the information should be read from the manifest during the build.  I think either way would work, and it's a trade off between where you have complexity.

Reply:
Yeah, I'm not a fan of the duplication as it allows things to get out of sync and means the manifest is no longer a means of composition.